test/e2e/snaps/security-alert-blockaid.spec.js

```diff
@@ -0,0 +1,210 @@
+            ],
+          },
+        ];
+
+        for (const config of testBenignConfigs) {
+          const { logExpectedDetail, method, params } = config;
+
+          // Send JSON-RPC request
+          const request = JSON.stringify({
+            jsonrpc: '2.0',
+            method,
+            params,
+          });
+          await driver.executeScript(
+            `window.transactionHash = window.ethereum.request(${request})`,
+          );
+
+          // Wait for confirmation pop-up
+          const windowHandles = await getWindowHandles(driver, 3);
+          await driver.switchToWindow(windowHandles.popup);
+
+          const isPresent = await driver.isElementPresent(bannerAlertSelector);
+          assert.equal(
+            isPresent,
+            false,
+            `Banner alert unexpectedly found. \nExpected detail: ${logExpectedDetail}`,
+          );
+
+          // Wait for confirmation pop-up to close
+          await driver.clickElement({ text: 'Reject', tag: 'button' });
+          await driver.switchToWindow(windowHandles.dapp);
+        }
+      },
+    );
+  });
+
+  /**
+   * Disclaimer: this test may be missing checks for some reason types. e.g. blur, domain, and failed
+   */
+  it('should show security alerts for malicious requests', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withNetworkController(mainnetProviderConfig)
+          .withPermissionControllerConnectedToTestDapp()
+          .withPreferencesController({
+            securityAlertsEnabled: true,
+          })
+          .build(),
+        title: this.test.title,
+      },
+
+      async ({ driver }) => {
+        await driver.navigate();
+        await unlockWallet(driver);
+        await openDapp(driver);
+
+        const expectedTitle = 'This is a deceptive request';
+
+        // Configs sorted by expectedReason
+        const testMaliciousConfigs = [
+          {
+            expectedDescription:
+              'If you approve this request, a third party known for scams might take all your assets.',
+            expectedReason: 'permit_farming',
+            method: 'eth_signTypedData_v4',
```

Comment:
thinking, could we just click on the malicious signatures from the test dapp directly, instead of triggering the request using the console? The benefit would be that the test would be cleaner (less data, only one click) :thinking: what do you think?

Reply:
I wonder if you were thinking we could add each of the malicious req/res to the test dapp? I can be in for the approach you are suggesting. I'm thinking 

#### If we test some cases in test dapp and some with script requests:

pros: 
- test-dapp can make it cleaner and easier for a few tests (whichever ones we support)
- easier to add more cases using script requests

cons:
- coupled with test-dapp, though we do already do this for other tests
- inconsistency if we test both test-dapp + request through script

#### If we do add all cases to test dapp:

pros: 
- nice to support more cases in test-dapp
- less pollution in e2e as we don't need to mock requests
- cleaner / less setup as you mentioned

cons:
- coupled with test-dapp, though we do already do this for other tests
- might want to be careful taking too much test dapp space
- adding e2e test cases may require 2 repo updates unless

Reply:
Thanks @seaona

PR created: https://github.com/MetaMask/test-dapp/pull/258

interesting that I'm receiving a benign response from PPOM for the existing malicious ERC20 transfer button, and the newer one I've just created for malicious ERC721 transfer button. not sure why yet

Updated these tests to utilize the test dapp buttons. I think we can still move forward with merging this PR to test signatures as the appropriate buttons already exist in the test dapp